### PR TITLE
[WIP] Add support for standard* things

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

File: src/Microsoft.Dnx.Host.CoreClr/DomainManager.cs [hunk 1]
```diff
@@ -9,6 +9,8 @@
 using Microsoft.Dnx.Runtime;
 using Microsoft.Extensions.PlatformAbstractions;
 using Microsoft.Dnx.Runtime.Common.Impl;
+using System.IO;
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Host.CoreClr/DomainManager.cs]: sort usings

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/DotNetGenerationMapping.cs [hunk 2]
```diff
@@ -9,40 +9,53 @@ internal static class DotNetGenerationMapping
         private static readonly Version _version45 = new Version(4, 5);
         private static readonly Dictionary<FrameworkName, Version> _generationMappings = new Dictionary<FrameworkName, Version>()
         {
+            { new FrameworkName(VersionUtility.NetStandardAppFrameworkIdentifier, new Version(1, 0)), new Version(1, 0) },
+            { new FrameworkName(VersionUtility.NetStandardAppFrameworkIdentifier, new Version(1, 1)), new Version(1, 1) },
+            { new FrameworkName(VersionUtility.NetStandardAppFrameworkIdentifier, new Version(1, 2)), new Version(1, 2) },
+            { new FrameworkName(VersionUtility.NetStandardAppFrameworkIdentifier, new Version(1, 3)), new Version(1, 3) },
+            { new FrameworkName(VersionUtility.NetStandardAppFrameworkIdentifier, new Version(1, 4)), new Version(1, 4) },
+
+            // dnxcore50
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/DotNetGenerationMapping.cs]: fix the comments

File: test/Microsoft.Dnx.Runtime.Tests/VersionUtilityFacts.cs [hunk 3]
```diff
@@ -13,6 +13,30 @@ namespace Microsoft.Dnx.Runtime.Tests
     public class VersionUtilityFacts
     {
         [Theory]
+
+        [InlineData("netstandard1.4", "dotnet5.5", false)]
```
Comment (davidfowl) [inline test/Microsoft.Dnx.Runtime.Tests/VersionUtilityFacts.cs]: these need to be true

**Comment by pakrym** [comment]
:up: :date: added transitive fallbacks

File: src/Microsoft.Dnx.Runtime/NuGet/Utility/DotNetGenerationMapping.cs [hunk 4]
```diff
@@ -56,53 +67,70 @@ public static IEnumerable<FrameworkName> Expand(FrameworkName input)
             //  however, we still want direct DNX == DNX matches, so we do this ONLY if the identifiers don't already match
             // This also handles .NET Generation mappings
 
-            yield return input;
-
-            var gen = GetGeneration(input);
 
-            // dnxN -> netN -> dotnetY
-            if (input.Identifier.Equals(VersionUtility.DnxFrameworkIdentifier))
+            var candidates = new Queue<FrameworkName>();
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/DotNetGenerationMapping.cs]: bfs eh

Comment (pakrym) [inline src/Microsoft.Dnx.Runtime/NuGet/Utility/DotNetGenerationMapping.cs]: Yep, looks less cool than goto )

**Comment by davidfowl** [comment]
Make sure you do some tests with this and the .NET CLI, see how those things play together.

**Comment by anurse** [comment]
If things are mostly playing well together, this looks good to me :shipit:. It's not like this has to be super clean :).

**Comment by pakrym** [comment]
Hopefully fixed runtime.json issue

**Comment by anurse** [comment]
Looks good. Let's get it in. :shipit:

**Comment by PinpointTownes** [comment]
I'd bet there's a bug in the restore stack, because `dnu restore` is unable to resolve the transient dependencies when using `netstandard` or `netstandardapp`. When using `dotnet restore`, they are correctly resolved.

A likely related symptom is that the .nuspec file stored in each "dependency folder" contains `<group targetFramework="Unsupported0.0">` instead of `<group targetFramework=".NETStandard1.3">`.

**Pull request merged**